File: app/src/main/java/com/timewasteanalyzer/usage/model/AppUsage.java
package com.timewasteanalyzer.usage.model;


import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.graphics.drawable.Drawable;


public class AppUsage {

    private Drawable appIcon;

    private String appName;
    private String packageName;
    private Time timeInForeground;

    private int launchCount;
    private int percent;

    public AppUsage(Context context, String packageName) {
        final PackageManager pm = context.getPackageManager();
        ApplicationInfo ai;
        try {
            ai = pm.getApplicationInfo(packageName, 0);
            this.appName = (String) pm.getApplicationLabel(ai);
            this.appIcon = pm.getApplicationIcon(ai);

        } catch (final PackageManager.NameNotFoundException e) {

        }
        this.packageName = packageName;
    }

    public Time getTimeInForeground() {
        if (timeInForeground == null) {
            timeInForeground = new Time(0);
        }
        return timeInForeground;
    }

    public void increaseLaunchCount() {
        launchCount++;
    }

    public String getAppName() {
        return appName;
    }

    public String getPackageName() {
        return packageName;
    }

    public Drawable getAppIcon() {
        return appIcon;
    }

    public int getPercent() {
        //return percent;
        return 50;
    }

    public int getLaunchCount() {
        return launchCount;
    }
}

File: app/src/main/kotlin/com/timewasteanalyzer/usage/model/AppUsage.kt
package com.timewasteanalyzer.usage.model


import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.graphics.drawable.Drawable


class AppUsage(context: Context, val packageName: String) {

    var appIcon: Drawable? = null
        private set

    var appName: String? = null
        private set

    var msInForeground: Long = 0
        private set

    var launchCount: Int = 0
        private set

    var percent: Int = 0
        private set

    val foregroundTimeString: String
        get() {
            val seconds = (msInForeground / 1000).toInt() % 60
            val minutes = (msInForeground / (1000 * 60) % 60).toInt()
            val hours = (msInForeground / (1000 * 60 * 60) % 24).toInt()

            return String.format("%02d:%02d:%02d", hours, minutes, seconds)
        }

    init {
        // Determine application name and icon
        val pm = context.packageManager
        val ai: ApplicationInfo
        try {
            ai = pm.getApplicationInfo(packageName, 0)
            this.appName = pm.getApplicationLabel(ai) as String
            this.appIcon = pm.getApplicationIcon(ai)

        } catch (e: PackageManager.NameNotFoundException) {
            // TODO handle exception
        } catch (e: Exception) {
            // TODO handle exception
        }

    }

    fun increaseLaunchCount() {
        launchCount++
    }

    fun addTimeInForeground(diff: Long) {
        this.msInForeground += diff
    }

    fun updatePercentage(total: Long) {
        this.percent = (msInForeground * 100.0 / total + 0.5).toInt()
    }
}
